app/src/pages/inside/dashboardPage/dashboardModal/dashboardModal.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withModal } from 'components/main/modal';
+import { submitAddEditDashboardFormAction } from 'controllers/dashboard';
+import { AddEditModal } from './addEditModal';
+import { DeleteModal } from './deleteModal';
+
+@withModal('dashboardModal')
+@connect(null, {
+  submitAddEditDashboardForm: submitAddEditDashboardFormAction,
+})
+export class DashboardModal extends Component {
```

Comment:
This component looks unnecessary since you can just show any modal by its' id, so there's no need to make another common component with `type` property.